docs/quarantine.md

```diff
@@ -0,0 +1,76 @@
+# Test Quarantine
+
+The execution of tests on CI should be as deterministic as possible, besides
+potential issues coming from infrastructure or external dependencies. When a
+test sometimes fails and sometimes passes, apparently randomly, without any
+relationship with the changes in a PR, it has very negative implications:
+* The results of the whole suite are less valuable, we can no longer trust
+the information given by the suite execution. Changes are added to the codebase
+without actually knowing if they are going to break something.
+* As a consequence, we can't be completely sure about the state of the product
+at release time, it can be ok or not.
+* The development process is affected negatively. Developers need to wait
+until a lucky execution aligns all the required tests in green, which can take
+more than a week in some cases.
+
+In summary, the CI system stops helping the product evolution and instead becomes
+an obstacle.
+
+We have introduced a quarantine methodology to put apart the tests that don't
+behave deterministically until they are fixed, so that we can keep the rest of
+the suite as healthy as possible. You can read more about test quarantine
+methodology in [1] and [2], and about an actual implementation in [3].
+
+## Procedure
+
+In order to remove as much as possible the influence of changes in PRs to
+determine the stability of the suite, we will take into account only results
+from the periodics that run e2e tests from master, these jobs can be checked
+[on testgrid].
+
+In the following definitions we will consider test failures only when reported on
+testgrid in jobs where less than 5 tests failed, so that we don't take into
+account systemic failures caused for instance by an infrastructure problem.
+
+### Putting tests in quarantine
+
+A test must be put in quarantine when any of these conditions is met:
+* It has failed 2 times or more in the last two weeks.
+* It has failed 5 times or more in the last 3 days.
+
+A PR will be proposed on Mondays every two weeks with a batch of the tests that
+met the first condition. A PR can be proposed at any time for the tests that meet
+the second condition. In both cases the PR will add the text `[QUARANTINE]` to
+each test's description in the code. An email will be sent to the owners of the
+suspected tests.
+
+Each quarantined test must have a team owner. The PR will add the text
+`[owner:@sig-{compute,network,storage}]` to each test's description.
```

Comment:
How about we create teams for these sigs here: https://github.com/kubevirt/project-infra/blob/master/github/ci/prow-deploy/files/orgs.yaml#L651
This would have the advantage that we could assign the team directly on an issue or pr?

Reply:
Great idea, this would make things much easier for tracking on the gh side, we can assign the issues from flake-issue-creator to these teams.

Reply:
Makes sense to me, updated.